packages/trip-form/src/__mocks__/modes-en.ts

```diff
@@ -194,4 +194,27 @@ const commonModes = {
   ]
 };
 
+// This variant of the data adds an option with a company as the first
+// option in the "CAR" category that has mixed modes ("CAR_PARK", "CAR_HAIL")
+export const modesWithCompanyFirstMixedCategory = {
+  ...commonModes,
+  categories: commonModes.categories.map(c => {
+    const nextCategory = { ...c };
+
+    if (c.id === "CAR") {
+      nextCategory.options = [
+        {
+          company: "LYFT",
+          mode: "CAR_HAIL",
+          label: "Lyft",
+          url: "https://www.lyft.com"
+        },
+        ...nextCategory.options
+      ];
```

Comment:
It looks like array spreading does not play well with TypeScript... You could try (assuming `nextCategory` is a deep clone of argument `c`):
```js
      nextCategory.options.splice(0, 0, {
          company: "LYFT",
          label: "Lyft",
          mode: "CAR_HAIL",
          url: "https://www.lyft.com"
      });

```

Reply:
typescript raises the same error when I switch to the `splice` syntax, I think the problem is that it's validating the option object against [`FullModeOption`](https://github.com/opentripplanner/otp-ui/blob/master/packages/trip-form/src/types.ts#L25) (from `src/types.ts`) instead of [`ModeOption`](https://github.com/opentripplanner/otp-ui/blob/master/packages/trip-form/src/TripOptions/types.d.ts#L3) (from `src/TripOptions/types.d.ts`).  Do you know of a way to control the association from the data to the type in this context?